File: Search-TDD-CSharp/Program/CommandController/CommandParser.cs
using Program.View;
using Search.Index;
using Search.Search;

namespace Program.CommandController
{
    public class CommandParser : ICommandParser
    {

        private const string ReadCommand = "read";
        private const string SearchCommand = "search";
        private const char WhiteSpace = ' ';

        private readonly IIndexer _indexer;
        private readonly ISearcher _searcher;

        public CommandParser(IIndexer indexer, ISearcher searcher)
        {
            _indexer = indexer;
            _searcher = searcher;
        }
        
        public void ParseCommand(string command)
        {
            string parsedCommand = command.Substring(command.IndexOf(WhiteSpace) + 1);
            if (command.StartsWith(ReadCommand))
            {
                ReadExecute(parsedCommand);
            }
            else if (command.StartsWith(SearchCommand))
            {
                SearchExecute(parsedCommand);
            }
        }

        private void ReadExecute(string folderName)
        {
            _indexer.Index(folderName.Trim());
        }

        private void SearchExecute(string command)
        {
            var searchResult = _searcher.Search(command);
            Menu.ShowResult(searchResult);
        }
    }
}

File: search-TDD/src/main/java/controller/command/CommandParser.java
package controller.command;

import controller.IndexController;
import controller.ProgramController;
import controller.searcher.Searcher;
import view.Menu;

import java.util.HashSet;

public class CommandParser {

    private final static String READ = "read";
    private final static String SEARCH = "search";
    private final static char WHITE_SPACE = ' ';

    public void chooseCommand(String command) {
        String parsedCommand = command.substring(command.indexOf(WHITE_SPACE) + 1);
        if (command.startsWith(READ)) {
            readCommand(parsedCommand);
        } else if (command.startsWith(SEARCH)) {
            searchCommand(parsedCommand);
        }
    }

    private void readCommand(String folderName) {
        ProgramController controllerInstance = ProgramController.getInstance();
        IndexController indexController = controllerInstance.getIndexController();
        indexController.addFolderToDatabase(folderName);
    }

    private void searchCommand(String command) {
        ProgramController controllerInstance = ProgramController.getInstance();
        Searcher searcher = controllerInstance.getSearcher();
        HashSet<String> searchResult = searcher.search(command);
        Menu.showResult(searchResult);
    }

}
